otx/cli/manager/config_manager.py

```diff
@@ -243,6 +241,96 @@ def auto_task_detection(self, data_roots: str) -> str:
         self.data_format = self.dataset_manager.get_data_format(data_roots)
         return self._get_task_type_from_data_format(self.data_format)
 
+    def _configure_train_type(self):
+        """Auto train type detection.
+
+        If self.args.train_type is not None -> use args.train_type
+        If train_data_roots contains only set of images -> Self-SL
+        If unlabeled-data-roots were passed to CLI -> use Semi-SL
+        If unlabeled_images presented in dataset structure and it is sufficient to start Semi-SL -> Semi-SL
+        Overwise set Incremental training type.
+        """
+
+        def _check_is_only_images(dir):
+            """Check if a directory contains only images."""
+            import glob
+
+            pngs = glob.glob(f"{dir}/*.png")
+            jpgs = glob.glob(f"{dir}/*.jpg")
+            return len(pngs) > 0 or len(jpgs) > 0
+
+        def _check_semisl_requirements(train_dir, unlabeled_dir, thershold=0.07):
+            """Check if quantity of unlabeled images is sufficient for Semi-SL learning."""
+            all_unlabeled_images = len(os.listdir(unlabeled_dir))
+            if all_unlabeled_images <= 1:
+                # the folder is empty or only 1 image there
+                return False
+            all_train_images = len(os.listdir(train_dir))
+            # check if number of unlabeled images is more than relative thershold
+            if all_unlabeled_images >= thershold * all_train_images:
+                return True
+            return False
+
+        # if user explicitly passed train type via args
+        if self.args.train_type is not None:
+            self.train_type = self.args.train_type
+            return
+
+        path_to_train_data = Path(self.args.train_data_roots)
+        if not Path.is_dir(path_to_train_data):
+            raise ValueError(
+                "train-data-roots isn't a directory or it doesn't exist. "
+                "Please, check command line and directory path."
+            )
+
+        if _check_is_only_images(path_to_train_data):
+            # If train folder with images only was passed to args
+            # Then we start self-supervised training
+            print("[*] Selfsupervised training type detected")
+            self.train_type = "Selfsupervised"
+            return
+
+        # if user explicitly passed unlabeled images folder
+        if self.args.unlabeled_data_roots is not None:
+            unlabeled_path = Path(self.args.unlabeled_data_roots)
+            if Path.is_dir(unlabeled_path):
+                if not _check_semisl_requirements(path_to_train_data, unlabeled_path):
+                    print(
```

Comment:
In this case, isn't the code to update to Incremental missing?

Reply:
Yes, you are right, thanks
Need to check tests as well, why this case wasn't caught

Reply:
For the warning message, I think using a logger is a more appropriate way since users could check the .log file whenever they want to see. Although .log file is not automatically saved now, I hope it will be fixed later.

Reply:
I added logging.warning instead of print